core/src/core_tests/workflow_tasks.rs

```diff
@@ -2856,13 +2913,14 @@ async fn sets_build_id_from_wft_complete() {
 
     worker.register_wf(DEFAULT_WORKFLOW_TYPE, |ctx: WfContext| async move {
         // First task, it should be empty, since replaying and nothing in first WFT completed
-        ctx.timer(Duration::from_secs(1)).await;
         assert_eq!(ctx.current_build_id(), None);
         ctx.timer(Duration::from_secs(1)).await;
         assert_eq!(ctx.current_build_id(), Some("enchi-cat".to_string()));
         ctx.timer(Duration::from_secs(1)).await;
         // Not replaying at this point, so we should see the worker's build id
         assert_eq!(ctx.current_build_id(), Some("fierce-predator".to_string()));
+        ctx.timer(Duration::from_secs(1)).await;
+        assert_eq!(ctx.current_build_id(), Some("fierce-predator".to_string()));
```

Comment:
This test was just actually wrong before. All the assertions should've been before rather than after the timer.